plugins/filter/to_prettytable.py

```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025, Timur Gadiev <tgadiev@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+
+DOCUMENTATION = '''
+name: to_prettytable
+short_description: Format a list of dictionaries as an ASCII table
+version_added: "10.7.0"
+author: Timur Gadiev (@tgadiev)
+description:
+- This filter takes a list of dictionaries and formats it as an ASCII table using the I(prettytable) Python library.
+requirements:
+- prettytable
+options:
+  _input:
+    description: A list of dictionaries to format.
+    type: list
+    elements: dictionary
+    required: true
+  column_order:
+    description: List of column names to specify the order of columns in the table.
+    type: list
+    elements: string
+  header_names:
+    description: List of custom header names to use instead of dictionary keys.
+    type: list
+    elements: string
+  column_alignments:
+    description:
+    - >-
+        Dictionary where keys are column names and values are alignment settings.
+        Valid alignment values are C(left), C(center), C(right), C(l), C(c), or C(r).
+    - >-
+        For example, V({'name': 'left', 'id': 'right'}) will align the C(name) column to the left
+        and the C(id) column to the right.
+    type: dictionary
+'''
+
+EXAMPLES = '''
+---
+- name: Display a list of users as a table
+  vars:
+    users:
+      - name: Alice
+        age: 25
+        role: admin
+      - name: Bob
+        age: 30
+        role: user
```

Comment:
The consistency pedantic strikes again 
```suggestion
    - name: Alice
      age: 25
      role: admin
    - name: Bob
      age: 30
      role: user
```

Reply:
Sorry, what consistency are we talking about here? This is an example of an ansible code in yaml format. And according to yamllint rules (used by ansible-lint for example) block sequences should be indented by default. See here for example:
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation

I'm not against this change, but I'm really confused and I don't understand why we need it. I checked other filter plugins code and I can see the same indentation as mine there. For example, `lists_mergeby`.

Reply:
The indentation in the PR looks good to me, I'm not sure why it should be changed. The resulting indentation doens't look correct.

Reply:
First of all, I'm not holding this back because of indentation, I don't want to make this thing bigger than it is. That being said, I apologise for not being very clear. The consistency difference I was referring to is the one between a list like lines 15-16
```yaml
requirements:
- prettytable
```
and lines 46-47
```yaml
    users:
      - name: Alice
```
The `-` is aligned with the containing element in one, whereas in the other it is indented 2 spaces.

Again, this is a very tiny nit, this is not holding back the merge, I just wanted to explain I wrote that before.

Reply:
In that case lines 15-16 should be fixed, IMO, as generally the indentation in this repo is matching the quoted block, but not lines 15-16.

Reply:
Fixed

Reply:
Thanks!